### DISCUSSION TITLE
Add a DatePeriod facade (as replacement for CarbonPeriod)

### DISCUSSION THREAD

**Discussion by halloei** [laravel/framework#40302] [Ideas]
I've posted this 3 years ago in ideas (https://github.com/laravel/ideas/issues/1546):

As I'd like to work with immutable dates in my project, I've added `\Illuminate\Support\Facades\Date::use(\Carbon\CarbonImmutable::class);` to my AppServiceProvider. In that way, Laravel does always return an instance of `CarbonImmutable` instead of `Carbon` (i.e. when using `now()`).

But of course this doesn't change the items of `\Carbon\CarbonPeriod` to `CarbonImmutable`.

So, as we have a `Date` facade, what about a `DatePeriod` facade, that respects the setting of `Date::use()`?

**Comment by kylekatarnls** [comment]
Hello.

> But of course this doesn't change the items of \Carbon\CarbonPeriod to CarbonImmutable.

For the record, in Carbon v3, when creating a `CarbonPeriod` from `CarbonImmutable` instance(s), it will by default return a `CarbonPeriodImmutable` and so its items will be `CarbonImmutable`.

And currently you can already create explictly `CarbonPeriodImmutable` periods:

```php
$period = CarbonPeriodImmutable::create('Today', '6 hours', 'Tomorrow');
var_dump(array_map(get_class(...), $period->toArray()));
```
All the items here in the array are `Carbon\CarbonImmutable`.

You can get the same result starting from a `CarbonPeriod` object:
```php
$period = CarbonPeriod::create('Today', '6 hours', 'Tomorrow')->setDateClass(CarbonImmutable::class);
var_dump(array_map(get_class(...), $period->toArray()));
```
